contracts/eip/queryable/ERC721AQueryable.sol

```diff
@@ -0,0 +1,168 @@
+// SPDX-License-Identifier: MIT
+// ERC721A Contracts v3.3.0
+// Creator: Chiru Labs
+
+pragma solidity ^0.8.4;
+
+import "./IERC721AQueryable.sol";
+import "../ERC721AVirtualApprove.sol";
+
+/**
+ * @title ERC721A Queryable
+ * @dev ERC721A subclass with convenience query functions.
+ */
+abstract contract ERC721AQueryable is ERC721A, IERC721AQueryable {
```

Comment:
don't we already have this file somewhere?

Reply:
> don't we already have this file somewhere?

There isn't a non-upgradable version currently.

Reply:
Gotcha